### PR TITLE
#4792 Shabondi service document

### PR DESCRIPTION
**Pull request by vitojeng**
### Checklist
- [x] [Review Contribution Guidelines](https://ohara.readthedocs.io/en/latest/contributing.html)
- [x] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#ohara-team))
- [x] Add Assignees (of course, you are the assignee by default!)
- [x] DON'T set a Milestone
- [x] DON'T add any labels (PR labels will be added automatically by Jenkins)
- [x] Link to an issue(You can do so by using the required by keyword: required by #4792)
- [ ] Add unit tests (or please explain why this PR is not worth having new tests)
- [ ] Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. If the failed tests are unrelated, you can add a comment in the PR thread)

### CODE REVIEW

File: docs/src/contrib-docs/quick_example.rst [hunk 1]
```diff
@@ -684,6 +684,20 @@ Http Restful
...
+  .. code-block:: text
+
+    .. code-block:: http
+
+      POST /blog/post.cgi HTTP/1.1
+      Host: www.example.com:443
+      Date: Mon, 23 May 2005 22:38:34 GMT
+      Content-Type: application/json; charset=UTF-8
+
+      {"id": 1,
```
Comment (konekoya) [inline docs/src/contrib-docs/quick_example.rst:696]: The format is a bit off?

File: docs/src/shabondi.rst [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+On the other hand, you can send the http request to **Shabondi sink** service to fetch the output data of the Pipeline.
+
+Following is a simple diagram of Pipeline to demonstrate about both the source and sink of Shabondi:
+  .. figure:: images/shabondi-pipeline.png
+     :alt: Shabondi in the pipeline
+
+Data format
+------------
+
+Both Shabondi source and sink use :ref:`Row <connector-datamodel>` in json format for data input and output.
```
Comment (konekoya) [inline docs/src/shabondi.rst:34]: json -> _JSON_

File: docs/src/shabondi.rst [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+  +---------+-------+-----+-------------------+---------+
+  |         | name  | age | email             | career  |
+  +---------+-------+-----+-------------------+---------+
+  | (row 1) | jason | 42  | jason@example.com | Surgeon |
+  +---------+-------+-----+-------------------+---------+
+  | (row n) | ...   | ... | ...               | ...     |
+  +---------+-------+-----+-------------------+---------+
+
+
+Example of row using java:
```
Comment (konekoya) [inline docs/src/shabondi.rst:49]: java -> _Java_

File: docs/src/contrib-docs/quick_example.rst (hunk 1 above)
Comment (vitojeng) [reply docs/src/contrib-docs/quick_example.rst:696]: You mean json format?  The data of json from L696 to L699.
The whole code-block is a example of **http response**.

File: docs/src/shabondi.rst [hunk 4]
```diff
@@ -0,0 +1,221 @@
...
+
+Following is a simple diagram of Pipeline to demonstrate about both the source and sink of Shabondi:
+  .. figure:: images/shabondi-pipeline.png
+     :alt: Shabondi in the pipeline
+
+Data format
+------------
+
+Both Shabondi source and sink use :ref:`Row <connector-datamodel>` in json format for data input and output.
+Row is a table structure data defined in Ohara code base. A row is comprised of multiple cells. Each cell has its **name** and **value**.
```
Comment (konekoya) [inline docs/src/shabondi.rst:35]: I feel like we can omit _data_ and so -> "Row is a table structure" or "Row is a table like data structure" :)

File: docs/src/shabondi.rst [hunk 5]
```diff
@@ -0,0 +1,221 @@
...
+
+* Ohara Manager: (TBD)
+* Use Configurator :ref:`REST API <rest-shabondi>` to create and start Shabondi service.
+
+After the deployment and start up is completed, the Shabondi service is ready.
+
+Source service API
+---------------------
+
+Shabondi source service receive the message of single row from Http request and then
```
Comment (konekoya) [inline docs/src/shabondi.rst:91]: receive -> receives

Comment (konekoya) [reply docs/src/shabondi.rst:91]: I'd probably rephrase this to:

Shabondi source service receives single row message through HTTP requests and then writes to the connected...

File: docs/src/shabondi.rst [hunk 6]
```diff
@@ -0,0 +1,221 @@
...
+write the row into the connected **Topic**.
+
+Send Row
+^^^^^^^^
+
+Send a json data of :ref:`Row <connector-datamodel>` to Shabondi source service.
```
Comment (konekoya) [inline docs/src/shabondi.rst:97]: json -> _JSON_

File: docs/src/contrib-docs/quick_example.rst (hunk 1 above)
Comment (konekoya) [reply docs/src/contrib-docs/quick_example.rst:696]: Yep, i think the `id:1` on line 696 should have a newline :)

File: docs/src/shabondi.rst (hunk 6 above)
Comment (konekoya) [reply docs/src/shabondi.rst:97]: Same for the rest docs

File: docs/src/shabondi.rst [hunk 7]
```diff
@@ -0,0 +1,221 @@
...
+
+Deployment
+-----------
+
+There are two ways to deploy Shabondi service
+
+* Ohara Manager: (TBD)
+* Use Configurator :ref:`REST API <rest-shabondi>` to create and start Shabondi service.
+
+After the deployment and start up is completed, the Shabondi service is ready.
```
Comment (konekoya) [inline docs/src/shabondi.rst:86]: Not sure I understand this right or not...
"After a Shabondi service is properly configured, deploy and running. It's ready to receive and send requests via HTTP"

Comment (vitojeng) [reply docs/src/shabondi.rst:86]: Nice. Thanks a lot.

File: docs/src/contrib-docs/quick_example.rst (hunk 1 above)
Comment (vitojeng) [reply docs/src/contrib-docs/quick_example.rst:696]: Done.

File: docs/src/shabondi.rst (hunk 2 above)
Comment (vitojeng) [reply docs/src/shabondi.rst:34]: Done.

File: docs/src/shabondi.rst (hunk 3 above)
Comment (vitojeng) [reply docs/src/shabondi.rst:49]: Done.

File: docs/src/shabondi.rst (hunk 4 above)
Comment (vitojeng) [reply docs/src/shabondi.rst:35]: Awesome!

File: docs/src/shabondi.rst (hunk 5 above)
Comment (vitojeng) [reply docs/src/shabondi.rst:91]: Done.

**Comment by konekoya** [comment]
retry

**Comment by vitojeng** [comment]
@konekoya Sorry forgot to submit PR.

**Comment by chia7712** [comment]
retry
